unidist/core/backends/mpi/core/controller/api.py

```diff
@@ -301,6 +305,7 @@ def wait_impl(data_id):
 
         operation_type = common.Operation.WAIT
         operation_data = {"id": data_id.base_data_id()}
+        # a blocking send is used because completion is necessary for processing to continue
```

Comment:
```suggestion
        # We use a blocking send here because we have to wait for
        # completion of the communication, which is necessary for the pipeline to continue.
```